6404, 6405 zvol test suite fixes

ahrens opened the pull request:
[6404](https://www.illumos.org/issues/6404) zvol_swap_006_pos can occasionally fail due to swaplen being < 16
[6405](https://www.illumos.org/issues/6405) zvol test setup is non deterministic

wca commented:
I made a lot of changes to the ZFS test suite (or rather, our fork of it) early this year trying to remove non-deterministic behavior.  IMHO, instead of using RANDOM, all tests should simply test all feasible cases.

ahrens commented:
@wca That makes sense.

ahrens commented:
@wca Can I count you as a reviewer of this code?

ahrens commented:
@zettabot go

wca commented:
LGTM

ahrens commented:
@zettabot go

ahrens committed:
6404 zvol_swap_006_pos can occasionally fail due to swaplen being < 16
6405 zvol test setup is non deterministic
Reviewed by: George Wilson <george.wilson@delphix.com>
Reviewed by: John Kennedy <john.kennedy@delphix.com>
Reviewed by: Will Andrews <will@freebsd.org>

ahrens commented:
@zettabot go

avg-I commented on usr/src/test/zfs-tests/tests/functional/zvol/zvol_swap/zvol_swap_004_pos.ksh:
```diff
@@ -51,7 +51,7 @@ typeset -i min max mem
-for vbs in 512 1024 2048 4096 8192 16384 32768 65536 131072; do
+for vbs in 8192 16384 32768 65536 131072; do
 	for multiplier in 1 32 16384 131072; do
 		((volsize = vbs * multiplier))
```
I solved this problem but skipping an iteration (using `continue`) if `volsize` is less than `2 * $(PAGESIZE)`.
At least that's how I interpreted information in the `swap` manual page. That seems to work.

avg-I replied:
BTW, do the `new_volsize` checks below actually work?
In my environment adding a volume as a swap device never changes its size, volsize always stays the same.

yuripankov commented on usr/src/test/zfs-tests/tests/functional/zvol/zvol_misc/zvol_misc_001_neg.ksh:
```diff
@@ -44,6 +44,8 @@
 
 verify_runnable "global"
 
+volsize=$(zfs get -H -o value volsize $TESTPOOL/$TESTVOL)
```
Here and below - please use $ZFS instead of 'zfs'.

yuripankov commented:
Matt, do you need help with this?

ahrens commented:
tests passed.  failure of zvol_swap/setup is a known issue, as is the ztest failure.  zvol_misc* tests now pass.

jwk404 commented:
LGTM.

WRT the comment about using zfs vs $ZFS (and likewise for other commands) I'd suggest not bothering with this for now. In the near future, we're going to change the tests to use the relative pathnames of commands. The method we're using now for allowing the suite to control what binaries are used puts too much of that burden on test developers. It's also an unproductive use of time for test writers and reviewers alike.

The same end result can be achieved by allowing writers to use the commands they're familiar with (cat, grep, rm, etc.) and enforcing a specific $PATH in the script that runs the suite. Alternatively, PATH could be set to one directory that contains symlinks to real commands. We'll be pursuing one of these options in the near future.

yuripv commented:
@jwk404 Not sure I follow - allowing test developers to rely on arbitrary features of commands outside of the system bin directories will make the suite inconsistent. What kind of burden does it put on test developers?
